metricbeat/scripts/docs_collector.py

```diff
@@ -19,20 +21,34 @@ def collect(beat_name):
 
     modules_list = {}
 
-    # Iterate over all modules
-    for module in sorted(os.listdir(base_dir)):
-
-        module_doc = path + "/" + module + "/_meta/docs.asciidoc"
+    modules_path = [{
+        'base_dir': oss_base_dir,
+        'path': oss_path + "/" + module,
+        'name': module,
+        'metricsets': sorted(os.listdir(oss_path + "/" + module))
+    } for module in filter(lambda module: os.path.isfile(oss_path + "/" + module + "/_meta/docs.asciidoc"),
+                           os.listdir(oss_base_dir))]
+
+    if os.path.isdir(os.path.abspath(xpack_base_dir)):
+        modules_path += [{
+            'base_dir': xpack_base_dir,
+            'path': xpack_path + "/" + module,
+            'name': module,
+            'metricsets': sorted(os.listdir(xpack_path + "/" + module)),
```

Comment:
[Named tuples](https://docs.python.org/2/library/collections.html#collections.namedtuple) are pretty handy in these cases, specially when fields of these objects are accessed so many times.